README.md

```diff
@@ -3,47 +3,81 @@
 
 
-## System-level configuration options
+## Field validation using conditional logic and regular expressions
 
-### Enable regular expression action tag
-Enable this to allow use of the **@REGEX** action tag.
+There are 3 action tags which can be used to provide additional field validation capabilities:
 
+* **@VALIDATE-LOGIC** &ndash; This can be used to supply conditional logic which must be valid in
+  order for the field value to be accepted.<br>
+  For example, if using an *email* and *confirm email* field, the confirm field could utilise the
+  action tag as follows:<br>
+  `@VALIDATE-LOGIC([email] = [email_confirm])`
+* **@VALIDATE-MESSAGE** &ndash; This can be used to supply a custom message to be displayed in
+  place of the standard message when a field validation error is triggered by the `@VALIDATE-LOGIC`
+  or `@REGEX` action tags.<br>
+  For example, a *confirm email* field might use:<br>
+  `@VALIDATE-MESSAGE="Sorry, the email addresses must match."`
+* **@REGEX** &ndash; This can be used to define a specific format as a regular expression which the
+  field value must satisfy.<br>
+  For example, an initials field which accepts 3 uppercase letters or 2 uppercase letters separated
+  by a hypthen could utilise the action tag as follows:<br>
+  `@REGEX="^[A-Z][A-Z-][A-Z]$"`<br>
 
-## Project-level configuration options
+**Please note:**
+* The *@REGEX* tag will only be available if it has been enabled in the module system settings.
+* The *@VALIDATE-LOGIC* and *@REGEX* action tags can be used simultaneously. A validation error will
+  result if either the logic or regular expression condition fails.
+* Blank fields will not be validated by these action tags. Set fields as *required* to prohibit
+  blank values.
 
-### Require field validation to pass on form submission
-This will check that when forms are submitted with status set to complete, that there are no field
-validation errors and no empty required fields. Submission will be blocked if there is a validation
-error or an empty required field.
+## Calculated default values
+
+Use the **@DEFAULT-CALC** action tag to provide a field with the result of a calculation as a
+default value.
+
+This action tag can be used in the same manner as the *@CALCTEXT* action tag but will populate the
+field similarly to the *@DEFAULT* action tag instead of creating a calculated field.
+
+**Please note:** The *@DEFAULT-CALC* action tag will only be available if it has been enabled in the
+module system settings.
+
+## Random numbers
+
+Use the **@RANDOMNUMBER** action tag to provide a field with a random number as a default value.
+
+The random numbers generated are cryptographically secure.
+
+The random number will be a decimal value between 0 and 1, unless used in the format
+`@RANDOMNUMBER(x,y)`, where `x` and `y` are two numbers, in which case the number returned will be
+an integer between the two numbers (inclusive).
```

Comment:
I have followed above with @RANDOMNUMBER(1, 4) in textbox with integer but it came out nothing.  Have I setup incorrectly?

Reply:
The `@RANDOMNUMBER` action tag is now fixed so it will work if there are spaces between the parentheses.